osu.Game/Rulesets/RulesetInfo.cs

```diff
@@ -24,6 +24,11 @@ public class RulesetInfo : RealmObject, IEquatable<RulesetInfo>, IRulesetInfo
 
         public string InstantiationInfo { get; set; } = string.Empty;
 
+        /// <summary>
+        /// A best effort sort ID which provides stable ordering and puts online rulesets before non-online rulesets.
+        /// </summary>
+        public int SortID => OnlineID >= 0 ? OnlineID : Math.Abs(ShortName.GetHashCode());
```

Comment:
This is so dodgy. Why not just implement `IComparable`/a comparer?

Reply:
Sure thing, will do that. Was my first thought but not sure if it would have unintended consequences somewhere. Are you okay with the `GetHashCode` usage, or any other suggestions?

Reply:
The `GetHashCode()` usage is my only concern with it.

Reply:
I've updated to use `IComparable`.